runtime-sdk/src/callformat.rs

```diff
@@ -275,9 +276,9 @@ pub fn encrypt_result_x25519_deoxysii<C: Context>(
         .write_u32::<BigEndian>(index.try_into().unwrap())
         .unwrap();
     nonce.extend(&[0, 0, 0]);
-    if ctx.is_simulation() {
-        // Randomize the lower-order bytes of the nonce to facilitate private queries.
-        OsRng.fill_bytes(&mut nonce[deoxysii::NONCE_SIZE - 3..]);
+    if CurrentState::with(|state| !state.env().is_execute()) {
```

Comment:
This used to only check if it's simulation, was previous check wrong? Or do we only end up here either in simulation or execute? (and not check?)

Reply:
The new check is _not_ execute (there is an `!`), so it covers simulation mode and be same as before, additionally covering check and pre-schedule (which should likely never reach this stage, but it seems like a good default to have).

Reply:
> additionally covering check and pre-schedule (which should likely never reach this stage, but it seems like a good default to have).

Yeah that's what i thought